AEPAssurance/Source/AssurancePresentationDelegate.swift

```diff
@@ -12,8 +12,60 @@
 
 import Foundation
 
+///
+/// AssurancePresentationDelegate to delegate the Assurance Presentation flow for both Pin and QuickConnect
+///
 protocol AssurancePresentationDelegate {
+    ///
+    /// Returns true if there is an open socket connection
+    ///
+    var isConnected: Bool { get }
+    
+    ///
+    /// Initializes the pin screen flow
+    ///
+    func initializePinScreenFlow()
+    
+    ///
+    /// Tells the conforming delegate that the pin screen connect button has been clicked
+    ///
+    /// - Parameter: pin `String`
+    ///
     func pinScreenConnectClicked(_ pin: String)
+    
+    ///
+    /// Tells the conforming delegate that the pin screen cancel button has been clicked
+    ///
     func pinScreenCancelClicked()
+    
+    ///
+    /// Tells the conforming delegate that the disconned button has been clicked
+    ///
     func disconnectClicked()
+    
+#if DEBUG
```

Comment:
Is my understanding that this is an extra layer of check on top of `startSession()` API gate to ensure any internal misuse, correct?

Reply:
@prudrabhat Yep!